test/cpp/api/init.cpp

```diff
@@ -0,0 +1,88 @@
+#include <gtest/gtest.h>
+
+#include <torch/nn/init.h>
+#include <torch/nn/modules/linear.h>
+
+#include <test/cpp/api/init_baseline.h>
+#include <test/cpp/api/support.h>
+
+#include <vector>
+
+void check_exact_values(
+    std::vector<torch::Tensor> parameters,
+    std::vector<std::vector<torch::Tensor>> expected_parameters) {
+  ASSERT_EQ(parameters.size(), expected_parameters.size());
+
+  for (size_t i = 0; i < parameters.size(); i++) {
+    auto layerParameters = parameters[i];
+    auto expectedLayerParameters = expected_parameters[i];
+
+    ASSERT_EQ(layerParameters.size(0), expectedLayerParameters.size());
+
+    for (size_t p = 0; p < layerParameters.size(0); p++) {
+      auto tensor = layerParameters[p];
+      auto expectedTensor = expectedLayerParameters[p];
+
+      ASSERT_TRUE(
```

Comment:
When using `ASSERT` in a loop I recommend outputting some information to know which loop iteration failed.  Otherwise Gtest doesn't give you this information. For example here, write `ASSERT_EQ(layerParameters.size(0), expectedLayerParameters.size()) << "parameter #" << i;` or something like that, and also for the next assert below